pkg/apis/core/v1alpha1/condition.go

```diff
@@ -17,181 +17,229 @@ limitations under the License.
 package v1alpha1
 
 import (
+	"sort"
+
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// ConditionType type for possible conditions the resource could be in.
+// A ConditionType represents a condition a resource could be in.
 type ConditionType string
 
-// Resource conditions.
+// Condition types.
 const (
-	// Pending means that the resource create request has been received and is waiting to be fulfilled.
-	Pending ConditionType = "Pending"
-	// Creating means that the resource create request has been accepted and the resource is in
-	// the process of being created.
-	Creating ConditionType = "Creating"
-	// Deleting means that the resource is in the process of being deleted.
-	Deleting ConditionType = "Deleting"
-	// Failed means that the resource is in a failure state, for example it failed to be created.
-	Failed ConditionType = "Failed"
-	// Ready means that the resource creation has been successful and the resource is ready to
-	// accept requests and perform operations.
+	// Ready managed resources are believed to be ready to handle work.
 	Ready ConditionType = "Ready"
+
+	// Synced managed resources are believed to be in sync with the Kubernetes
+	// resources that manage their lifecycle.
+	Synced ConditionType = "Synced"
+)
+
+// A ConditionReason represents the reason a resource is in a condition.
+type ConditionReason string
+
+// Condition reasons.
+const (
```

Comment:
General thought.

I was never :100: sure if using "shared" conditions was the right approach. 
While I agree that many (if not all) resources share the same base condition, at the same time I do think that resource conditions should reflect specific resource state, i.e. to be resource specific.

The way @negz formulated `const` conditions value make them better suitable for multiple resources. Whereas in the past we had, for example, a `Running` condition, which did not make sense for "not-runnable" resources like storage buckets. So I think this is an improvement. 

With that said, do we think this is the right direction, i.e. golbal conditions vs. resource specific ones?
Again, this is just a thought and not a challenge, since I do see some cases of reusability, but not :100: sure.

What do you think?

Reply:
Yeah I also wondered this. My suspicion is that these reasons will be appropriate for at least 80% of cases, especially our main functionality of managing cloud resources. I'm not sure they make as much sense in the edge cases like controllers that manage Workloads. I think I'll have a stronger opinion on this once I've attempted to adapt all of our existing controllers to use these conditions.

Given I suspect these reasons will be suitable for 80% of our controllers, perhaps we could consider an approach in which we default to using these shared conditions, but fall back to defining our own for controllers where they don't make sense?